source/operation.ts

```diff
@@ -137,7 +139,7 @@ export function search({
  * @return {Object}      API operation
  */
 export function update(
-  type: string,
+  type: EntityType,
   keys: string[] | string,
   data: any,
 ): UpdateOperation {
```

Comment:
we could pass `TEntityType` as a generic to UpdateOperation, and have it be automatically inferred from the `type` param. that could be useful in `session.update` i believe, as it then gets back an update operation with the entity type properly typed to its own TEntityType. 

(same goes for all operations)

Reply:
That is the way to go and something I considered. However, I felt that expanding the scope of this PR wasn't the best option. 
There are quite a few different places where we can be more specific in the typing when this is implemented, but I feel that is best handled as a focused effort after this is merged.

Reply:
Alright. With that in mind, I'll mark this comment as resolved.